rllib/utils/exploration/random_encoder.py

```diff
@@ -0,0 +1,338 @@
+                vectors in latent space.
+            encoder_net_config (Optional[ModelConfigDict]): Optional model
+                configuration for the encoder network, producing embedding
+                vectors from observations. This can be used to configure
+                fcnet- or conv_net setups to properly process any
+                observation space.
+            beta (float): Hyperparameter to choose between exploration and
+                exploitation.
+            beta_schedule (str): Schedule to use for beta decay, one of
+                "constant" or "linear_decay".
+            rho (float): Beta decay factor, used for on-policy algorithm.
+            k_nn (int): Number of neighbours to set for K-NN entropy
+                estimation.
+            random_timesteps (int): The number of timesteps to act completely
+                randomly (see [1]).
+            sub_exploration (Optional[FromConfigSpec]): The config dict for
+                the underlying Exploration to use (e.g. epsilon-greedy for
+                DQN). If None, uses the FromSpecDict provided in the Policy's
+                default config.
+
+        Raises:
+            ValueError: If the input framework is Torch.
+        """
+        if framework == "torch":
+            raise ValueError("This RE3 implementation does not support Torch.")
+        super().__init__(
+            action_space, model=model, framework=framework, **kwargs)
+
+        self.beta = beta
+        self.rho = rho
+        self.k_nn = k_nn
+        self.embeds_dim = embeds_dim
+        if encoder_net_config is None:
+            encoder_net_config = self.policy_config["model"].copy()
+        self.encoder_net_config = encoder_net_config
+
+        # Auto-detection of underlying exploration functionality.
+        if sub_exploration is None:
+            # For discrete action spaces, use an underlying EpsilonGreedy with
+            # a special schedule.
+            if isinstance(self.action_space, Discrete):
+                sub_exploration = {
+                    "type": "EpsilonGreedy",
+                    "epsilon_schedule": {
+                        "type": "PiecewiseSchedule",
+                        # Step function (see [2]).
+                        "endpoints": [
+                            (0, 1.0),
+                            (random_timesteps + 1, 1.0),
+                            (random_timesteps + 2, 0.01),
+                        ],
+                        "outside_value": 0.01,
+                    },
+                }
+            elif isinstance(self.action_space, Box):
+                sub_exploration = {
+                    "type": "OrnsteinUhlenbeckNoise",
+                    "random_timesteps": random_timesteps,
+                }
+            else:
+                raise NotImplementedError
+
+        self.sub_exploration = sub_exploration
+
+        # Creates modules/layers inside the actual ModelV2.
```

Comment:
nit picking.
it's not really "inside" ModelV2?
more like "Creates ModelV2 embedding module / layers"?